File: 24-swap-nodes-in-pairs/swap-nodes-in-pairs.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def swapPairs(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        t=head
        while (t and t.next):
            data=t.val
            t.val=t.next.val
            t.next.val=data
            t=t.next.next

        return head
        

File: 24-swap-nodes-in-pairs/swap-nodes-in-pairs.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapPairs(ListNode head) {
        // ListNode temp=head;
        // int d;

        // ListNode tadd=head;
        // while(temp!=null&& temp.next!=null){
        //     tadd=temp;
        //     temp=temp.next;
        //     temp.next=tadd;
        //     temp=temp.next.next;

        // }
        // return head;
        if(head==null||head.next==null) return head;

        ListNode temp=new ListNode();
        ListNode prev=temp;
        prev.next=head;
        while(prev.next!=null && prev.next.next!=null){
            ListNode p1=prev.next;
            ListNode p2=p1.next;
            p1.next=p2.next;
            p2.next=p1;
            prev.next=p2;
            prev=p1;
        }
        return temp.next;

    }
}